src/dsp_tools/commands/xmlupload/models/upload_state.py

```diff
@@ -6,7 +6,7 @@
 from dsp_tools.commands.xmlupload.upload_config import UploadConfig
 
 
-@dataclass(frozen=True)
+@dataclass
```

Comment:
**suggestion (non-blocking):** keep data immutable where possible  
rather than un-freezing this dataclass, I would prefer creating a copy of the object with the shortened list, using the `dataclasses.replace()` function